### ISSUE TITLE
Fix MATLAB tests to not rely on ordering of dictionaries

### ISSUE THREAD

**Issue by InsertCreativityHere** [zeroc-ice/ice#696]
In at least the `Ice/objects` (but possible others where we use dictionaries with complex key-types), we rely on dictionaries passed to operations to be in the same order as when we sent them:
https://github.com/zeroc-ice/ice/blob/6d73d9244f2c22455cf2a599ba3ce4088ec42a31/matlab/test/Ice/objects/AllTests.m#L261-L270
However, Ice makes no guarantee that it will preserve the order of dictionary entries, so this method of accessing them is flawed.

We should instead fix the tests to test dictionary entries by key, instead of by order.

**Comment by InsertCreativityHere** [comment]
Fixed by 779a0e5 on master and 75c59c8 on 3.7 respectively.

**Comment by externl** [comment]
This is still failing

```

To get started, type doc.
For product information, visit www.mathworks.com.
 
Error: File: C:\Users\vagrant\workspace\ice\master\matlab\test\Ice\objects\AllTests.m Line: 268 Column: 13
At least one END is missing: the statement may begin here.

Error in client (line 30)
    initial = AllTests.allTests(helper);

Error in runTest (line 28)
        client(varargin);
ERROR: MATLAB error Exit Status: 0x00000001
````
Looks like a missing `END`
